src/api/CoinBase.ts

```
@@ -25,35 +26,77 @@ const CURRENCY_MAPPING: Record<string, string> = {
+            if (response) {
+                log.error('Response error:', response.data, response.status);
+            }
+
+            return Promise.reject(error);
+        }
+    }
+};
+
 export default class CoinBase extends BaseExchangeApi {
-    constructor() {
-        super('CoinBase', CURRENCY_MAPPING, ['RIF']);
+    private readonly _httpClient: AxiosInstance;
+
+    constructor(loglLevel: LogLevelDesc = 'error') {
+        super('CoinBase', CURRENCY_MAPPING, ['RIF', 'RBTC']);
+        log.setLevel(loglLevel);
+
+        this._httpClient = axios.create();
+        this._httpClient.interceptors.response.use(
+            interceptors.logRequest.onResponse,
+            interceptors.logRequest.onError
+        );
     }
 
     async queryExchangeRate(
         sourceCurrency: string,
         targetCurrency: string
     ): Promise<BigNumber> {
         const upperCaseTargetCurrency = targetCurrency.toUpperCase();
-        const response: Response = await fetch(
-            `${URL}?currency=${sourceCurrency}`,
-            {
-                method: 'get'
-            }
-        );
+        let response: AxiosResponse;
 
-        if (!response.ok) {
-            const {
-                errors: [error]
-            }: CoinBaseErrorResponse = await response.json();
-            throw Error(
-                `CoinBase API status ${response.status}/${response.statusText}/${error.message}`
-            );
+        try {
+            response = await axios.get(`${URL}?currency=${sourceCurrency}`);
+        } catch (error: unknown) {
+            const { response, request } = error as AxiosError;
+            if (response) {
+                throw Error(
+                    `CoinBase API status ${response.status}/${response.statusText}`
+                );
+            }
+            if (request) {
+                throw Error('No response received from CoinBase API');
+            }
+            throw new Error('The request was not sent to the CoinBase API');
```

why not just let the interceptor handle it?
 In any case, is it your intention to hide the original error? Just because this catch block does not seem to expose it anywhere.

---

not an issue